alacritty/src/event.rs

```diff
@@ -1216,12 +1227,39 @@ impl input::Processor<EventProxy, ActionContext<'_, Notifier, EventProxy>> {
                             *self.ctx.dirty = true;
                         }
                     },
-                    WindowEvent::Ime(ime) => {
-                        if let Ime::Commit(text) = ime {
+                    WindowEvent::Ime(ime) => match ime {
+                        Ime::Commit(text) => {
+                            // Clear preedit.
+                            self.ctx.display.ime.set_preedit(None);
+
+                            // Don't suppress the chars coming from the IME commit.
+                            let suppress = mem::take(self.ctx.suppress_chars);
                             for ch in text.chars() {
-                                self.received_char(ch)
+                                self.received_char(ch);
                             }
-                        }
+                            *self.ctx.suppress_chars = suppress;
```

Comment:
Suppress chars would only ever be triggered by key bindings that were entered while the IME is active. The fact that you're looking out for this here means there's a chance keybindings might be read while IME is performed? Do we want this or do we want to disable bindings during IME?

It seems strange to temporarily disable suppression here.